overlord/devicestate/devicemgr.go

```diff
@@ -1091,6 +1091,30 @@ func (m *DeviceManager) ensureCloudInitRestricted() error {
 	return nil
 }
 
+// hasInstallDeviceHook returns whether the gadget has an install-device hook.
+// It can return an error if the device has no gadget snap
+func (m *DeviceManager) hasInstallDeviceHook(model *asserts.Model) (bool, error) {
+	var hasInstallDeviceHook bool
+
+	gadgetInfo, err := snapstate.CurrentInfo(m.state, model.Gadget())
+	if err != nil {
+		return false, fmt.Errorf("device is seeded in install mode but has no gadget snap: %v", err)
+	}
+	hasInstallDeviceHook = (gadgetInfo.Hooks["install-device"] != nil)
```

Comment:
nitpick:
```suggestion
	gadgetInfo, err := snapstate.CurrentInfo(m.state, model.Gadget())
	if err != nil {
		return false, fmt.Errorf("device is seeded in install mode but has no gadget snap: %v", err)
	}
	hasInstallDeviceHook := (gadgetInfo.Hooks["install-device"] != nil)
```

Reply:
Ah nice, yea I think I was a bit to verbose in my code split-out, I didn't notice this. Thank you!